irmago_test.go

```diff
@@ -985,6 +985,371 @@ func TestWizardFromScheme(t *testing.T) {
+			credid("scheme.issuer.e"): credtype("scheme.issuer.e",
+				"scheme.issuer.d",
+			),
+		},
+	}
+
+	wizard := IssueWizard{
+		ID: "testwizard",
+		Contents: IssueWizardContents{
+			{{
+				credwizarditem("scheme.issuer.c"),
+			}},
+			{{{
+				Type:   IssueWizardItemTypeWebsite,
+				Header: &TranslatedString{"en": "header"},
+				Label:  &TranslatedString{"en": "label"},
+				Text:   &TranslatedString{"en": "text"},
+				URL:    &TranslatedString{"en": "https://example.com"},
+			}}},
+			{{{
+				Type:       IssueWizardItemTypeCredential,
+				Credential: credidptr("scheme.issuer.e"),
+				Text:       &TranslatedString{"en": "custom description of credential e"},
+			}}},
+		},
+	}
+
+	err := wizard.Validate(conf)
+	require.Equal(t, "non-credential types in wizard testwizard should come last", err.Error())
+}
+
+func TestWizardComplexity(t *testing.T) {
+	conf := &Configuration{
+		CredentialTypes: map[CredentialTypeIdentifier]*CredentialType{
+			credid("scheme.issuer.a"): credtype("scheme.issuer.a"),
+			credid("scheme.issuer.b"): credtype("scheme.issuer.b",
+				"scheme.issuer.a",
+			),
+			credid("scheme.issuer.c"): credtype("scheme.issuer.c",
+				"scheme.issuer.a", "scheme.issuer.b",
+			),
+			credid("scheme.issuer.d"): credtype("scheme.issuer.d",
+				"scheme.issuer.a",
+			),
+			credid("scheme.issuer.e"): credtype("scheme.issuer.e",
+				"scheme.issuer.d",
+			),
+			credid("scheme.issuer.e"): credtype("scheme.issuer.e"),
+			credid("scheme.issuer.f"): credtype("scheme.issuer.f",
+				"scheme.issuer.a",
+			),
+			credid("scheme.issuer.g"): credtype("scheme.issuer.g",
+				"scheme.issuer.a", "scheme.issuer.b",
+			),
+			credid("scheme.issuer.h"): credtype("scheme.issuer.h",
+				"scheme.issuer.a",
+			),
+			credid("scheme.issuer.i"): credtype("scheme.issuer.i",
+				"scheme.issuer.d",
+			),
+		},
+	}
+
+	wizard := IssueWizard{
+		ID: "testwizard",
+		Contents: IssueWizardContents{
+			{{
+				credwizarditem("scheme.issuer.c"),
+			}},
+			{{
+				credwizarditem("scheme.issuer.d"),
+			}},
+			{{
+				credwizarditem("scheme.issuer.e"),
+			}},
+			{{
+				credwizarditem("scheme.issuer.f"),
+			}},
+			{{
+				credwizarditem("scheme.issuer.g"),
+			}},
+			{{
+				credwizarditem("scheme.issuer.h"),
+			}},
+			{{
+				credwizarditem("scheme.issuer.i"),
+			}},
+			{{{
+				Type:       IssueWizardItemTypeCredential,
+				Credential: credidptr("scheme.issuer.e"),
+				Text:       &TranslatedString{"en": "custom description of credential e"},
+			}}},
+			{{{
+				Type:   IssueWizardItemTypeWebsite,
+				Header: &TranslatedString{"en": "header"},
+				Label:  &TranslatedString{"en": "label"},
+				Text:   &TranslatedString{"en": "text"},
+				URL:    &TranslatedString{"en": "https://example.com"},
+			}}},
+		},
+	}
+
+	err := wizard.Validate(conf)
+	require.Equal(t, "wizard with wizard ID testwizard too complex", err.Error())
+}
+
+func TestIssueWizardItemValidation(t *testing.T) {
+	dependencies := CredentialDependencies{
```

Comment:
Apart from populating `Configuration.CredentialTypes` below, you don't seem to use `dependencies at all`. Would it not be better to remove this and populate with just the two credential types which are used in the test? Same goes for `TestIssueWizardFAQSummariesValidation` below.